### DISCUSSION TITLE
get_fd_waveform_from_td

### DISCUSSION THREAD

**Discussion by gianlucaguidi** [gwastro/pycbc#4144] [General]
Hi,

as I am getting an error from waveform.py: get_fd_waveform_from_td, I am trying to understand the logic and I am perplexed by:

 if tsamples < len(hp):
        raise ValueError("The frequency spacing (df = {}) is too low to "
                         "generate the {} approximant from the time "
                         "domain".format(params['delta_f'], params['approximant']))

hp.resize(tsamples)
hc.resize(tsamples)

Is it correct?

**Comment by ahnitz** [comment]
@gianlucaguidi Could you provide an minimal example that reproduces this?

**Reply by ahnitz** [reply]
@gianlucaguidi Let me expand on what the message is trying to say. Let me know if this makes sense to you. 

Internally we generate the waveform in the time domain. The waveform is initially a time series with a particular duration. Let's say 10s as a hypothetical. The frequency spacing of frequency series is directly related to the duration of the time domain version of the same data, delta_f = 1.0 / time_duration. So if you requested a delta_f of 0.2, that would only correspond to a time series of 5 seconds. That means we'd either need to truncate the time series (possibly cutting off  your signal) or perhaps wrap it around the array in some way. If you choose a delta_f < 1.0 / 10 we would just zeropad the array, which we consider innocuous since it does not change the signal / waveform content.

**Reply by ahnitz** [reply]
@gianlucaguidi So, is this occuring in a case where it should not? Or would you expect some other behavior? If so, what? We'd be happy to alter this if there is a way to do so sensibly. Perhaps add an option to explicitly allow truncating the start of signal? Would that be what one would expect?

**Reply by gianlucaguidi** [reply]
Hi Alex, 

yes, sorry, below the error which occurs when running banksim with a filter-signal-length = 256 and a time-domain approximant (SpinTaylorT5):

Traceback (most recent call last):
  File "/home/elisa.nitoglia/O4/O4-BANK-SSM_v1/banksim_SpinTaylorT5/scripts/pycbc_banksim", line 321, in <module>
    stilde = get_waveform(options.signal_approximant,
  File "/home/elisa.nitoglia/O4/O4-BANK-SSM_v1/banksim_SpinTaylorT5/scripts/pycbc_banksim", line 116, in get_waveform
    hp, hc = get_fd_waveform(wf_params, approximant=approximant,
  File "/home/elisa.nitoglia/.conda/envs/myclone-igwn-py39-20220707/lib/python3.9/site-packages/pycbc/waveform/waveform.py", line 608, in get_fd_waveform
    return wav_gen(**input_params)
  File "/home/elisa.nitoglia/.conda/envs/myclone-igwn-py39-20220707/lib/python3.9/site-packages/pycbc/waveform/waveform.py", line 663, in get_fd_waveform_from_td
    raise ValueError("The frequency spacing (df = {}) is too low to "
ValueError: The frequency spacing (df = 0.00390625) is too low to generate the SpinTaylorT5 approximant from the time domain

thanks

Gianluca
